velox/exec/Window.cpp

```diff
@@ -41,16 +41,94 @@ void initKeyInfo(
   }
 }
 
-void checkDefaultWindowFrame(const core::WindowNode::Function& windowFunction) {
-  VELOX_CHECK_EQ(
-      windowFunction.frame.type, core::WindowNode::WindowType::kRange);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.startType,
-      core::WindowNode::BoundType::kUnboundedPreceding);
-  VELOX_CHECK_EQ(
-      windowFunction.frame.endType, core::WindowNode::BoundType::kCurrentRow);
-  VELOX_CHECK_EQ(windowFunction.frame.startValue, nullptr);
-  VELOX_CHECK_EQ(windowFunction.frame.endValue, nullptr);
+vector_size_t findFrameEndPointUnboundedPreceding(
+    core::WindowNode::WindowType /*type*/,
+    vector_size_t /*functionNumber*/,
+    vector_size_t partitionStartRow,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t /*currentRow*/,
+    std::optional<vector_size_t> offsetValue) {
+  return partitionStartRow;
+}
+
+vector_size_t findFrameEndPointKPreceding(
+    core::WindowNode::WindowType type,
+    vector_size_t functionNumber,
+    vector_size_t partitionStartRow,
+    vector_size_t /*partitionEndRow*/,
+    vector_size_t currentRow,
+    std::optional<vector_size_t> offsetValue) {
```

Comment:
Why is this optional ? I think you did this to make the signatures consistent across the functions. In that case please add a debug assertion that offset is always passed.

Reply:
Yes, that is why the optional param was added, included an assertion in this function too, but it is not required since a check is already there in createWindowFunctions.